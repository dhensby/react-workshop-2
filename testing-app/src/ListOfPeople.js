import React, { Component } from 'react'
import PropTypes from 'prop-types'

class ListOfPeople extends Component {
  renderPeople() {
    return this.props.people.map(person => (
      <li key={person.id}>
        <div style={{color: person.colour}}>
          <p>Name: { person.name }</p>
        </div>
      </li>
    ))
  }

  render() {
    return (
      <div>
        <h2>List of Jack's Friends</h2>
        <ul>{ this.renderPeople() }</ul>
      </div>
    )
  }
}
ListOfPeople.propTypes = {
  people: PropTypes.array.isRequired,
}

export default ListOfPeople
