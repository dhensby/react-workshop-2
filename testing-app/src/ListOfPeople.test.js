import { mount } from 'enzyme'
import ListOfPeople from './ListOfPeople'
import React from 'react'
import renderer from 'react-test-renderer'

test('it renders the heading', () => {
  const wrapper = mount(<ListOfPeople people={[]} />)
  expect(wrapper.find('h2').text()).toEqual(`List of Jack's Friends`)
})

test('it matches the snapshot', () => {
  const tree = renderer.create(
    <ListOfPeople people={[
      { id: 1, name: 'Jack', colour: 'blue' },
    ]} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

