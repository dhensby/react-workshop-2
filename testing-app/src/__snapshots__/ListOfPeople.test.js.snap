exports[`test it matches the snapshot 1`] = `
<div>
  <h2>
    List of Jack\'s Friends
  </h2>
  <ul>
    <li>
      <div
        style={
          Object {
            "color": "blue",
          }
        }>
        <p>
          Name: 
          Jack
        </p>
      </div>
    </li>
  </ul>
</div>
`;
